### PR TITLE
feat: [OI-245] linting & formatting

### PR DESCRIPTION
**Pull request by mmoio**
-- updates eslint dependencies
-- updates prettier
-- fixes and refactors eslint configuration
-- sctrictier lint checking and formatting
-- fixes linting problems
-- some refactoring

### CODE REVIEW

File: src/oneid/oneid-ecs-core/src/main/webui/Dangerfile.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
 // import custom DangerJS rules
```
Comment (sebbalex) [inline src/oneid/oneid-ecs-core/src/main/webui/Dangerfile.ts:1]: Danger is actually not used here, I think we can remove it

File: src/oneid/oneid-ecs-core/src/main/webui/eslint.config.mjs [hunk 2]
```diff
@@ -1,179 +1,53 @@
...
+  {
+    plugins: {
+      'react-hooks': reactHooks,
     },
+    rules: {
+      ...reactHooks.configs.recommended.rules,
+    },
+  },
+  {
+    files: ['*.mjs', '*.cjs', '*.js?(x)'],
```
Comment (sebbalex) [inline src/oneid/oneid-ecs-core/src/main/webui/eslint.config.mjs:50]: shouldn't we include also `*.ts?(x)` files also

Comment (mmoio) [reply src/oneid/oneid-ecs-core/src/main/webui/eslint.config.mjs:50]: This elint configuration load the files included in the tsconfig.json by default, so they should be already icluded

File: src/oneid/oneid-ecs-core/src/main/webui/Dangerfile.ts (hunk 1 above)
Comment (mmoio) [reply src/oneid/oneid-ecs-core/src/main/webui/Dangerfile.ts:1]: So i can remove the file entirelly?

Comment (sebbalex) [reply src/oneid/oneid-ecs-core/src/main/webui/Dangerfile.ts:1]: yes, and so the `danger-plugin-digitalcitizenship` as well

Comment (mmoio) [reply src/oneid/oneid-ecs-core/src/main/webui/Dangerfile.ts]: I've removed the Dangerfile and dependencies

**Review by sebbalex — APPROVED** [review]

**Pull request merged**